File: backend/app/schemas/navigation.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field

class NavigationItem(BaseModel):
    label: str
    url: str
    enabled: bool = True
    display_order: int = 1
    target_blank: bool = False
    children: Optional[List["NavigationItem"]] = []

class NavigationMenuBase(BaseModel):
    location: str  # header, footer, mobile
    items: List[NavigationItem] = []

class NavigationMenuCreate(NavigationMenuBase):
    pass

class NavigationMenuUpdate(BaseModel):
    items: List[NavigationItem]

class NavigationMenuResponse(NavigationMenuBase):
    id: str
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

File: admin/lib/types/navigation.ts
export interface NavigationItem {
  id?: string;
  label: string;
  url: string;
  enabled: boolean;
  display_order: number;
}

export interface NavigationMenu {
  id?: string;
  location: "header" | "footer";
  items: NavigationItem[];
  updated_at?: string;
}
